### PR TITLE
feat(Bonus Pagamenti Digitali): [#175266102] Add summary (graphical and textual) for period transactions and amount component

### PR DESCRIPTION
**Pull request by fabriziofff**
# This pr depends on #2330

## Short description
This pr adds the `BpdSummaryComponent` that displays information related to the selected period and amount.

#### Description
A concise description of the component architecture:
- `ShadowBox`: A box with a shadow
- `BpdBaseShadowBoxLayout`: The layout of the graphical component
- `ProgressBar`: A simple progress bar
- `TransactionsGraphicalSummary`: A graphical component that represents graphically the transaction in the period
- `TextualSummary`: A summary with textual information related to the period and amount.
- `BpdSummaryComponent`: The general component that displays textual + graphical information.

Each component implements some rules in order to display the appropriate graphic state. 

#### Composition

`ShadowBox` <---use-- `BpdBaseShadowBoxLayout` + `ProgressBar`  <--use--  `TransactionsGraphicalSummary`

`ShadowBox` <---use-- `BpdBaseShadowBoxLayout`  <--use--  `TextualSummary`

 `TransactionsGraphicalSummary` + `TextualSummary` <--use-- `BpdSummaryComponent`

### `TransactionsGraphicalSummary` graphical states

`transactions === 0`        |  `transactions < minTransactionNumber`  |  `transactions >= minTransactionNumber`
:-------------------------:|:---------------:|:-----------------:
|  |  

### `BpdSummaryComponent` graphical states

`status === closed && totalCashback >= maxPeriodCashback && transactionNumber >= minTransactionNumber`        |  `status === closed && totalCashback < maxPeriodCashback && transactionNumber >= minTransactionNumber`  |  `status === closed && transactionNumber < minTransactionNumber`
:-------------------------:|:---------------:|:-----------------:
|  |  

`status === current && transactionNumber < minTransactionNumber && totalCashback > 0`     |  `status === closed && today <= endGracePeriod && today >= period.endDate`  | `status === inactive`
:-------------------------:|:-----------:|:-------------:
|  |  

## List of changes proposed in this pull request
- Added all the component previously described to display the summary.
- Added `blue` color to `H2` core typography component.
- Added `prefetchBpdData` saga (and related trigger action `bpdDetailsLoadAll` as single entrypoint to load all the informations related to bpd.
- Added `profileNameSelector` to select the name of the profile.

### CODE REVIEW

File: ts/features/bonus/bpd/screens/details/components/summary/BpdSummaryComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+const Content = (sd: SummaryData) => (
+  <View>
+    {sd.period.status === "Inactive" ? null : (
+      <TransactionsGraphicalSummary
+        transactions={sd.amount.transactionNumber}
+        minTransactions={sd.period.minTransactionNumber}
+      />
+    )}
+    <View spacer={true} />
+    <TextualSummary period={sd.period} amount={sd.amount} name={sd.name} />
```
Comment (Undermaken) [inline ts/features/bonus/bpd/screens/details/components/summary/BpdSummaryComponent.tsx:37]:
```suggestion
    <TextualSummary {...sd} />
```
Surely more concise but less readable
Wht's your opinion?

File: ts/features/bonus/bpd/screens/details/components/summary/TextualSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+ */
+const chooseTextualInfoBox = (props: Props) => {
+  switch (props.period.status) {
+    case "Inactive":
+      return <InactivePeriod period={props.period} />;
+    case "Closed":
+      const today = new Date();
+      const endGracePeriod = new Date();
+      endGracePeriod.setDate(
+        props.period.endDate.getDate() + props.period.gracePeriod
```
Comment (Undermaken) [inline ts/features/bonus/bpd/screens/details/components/summary/TextualSummary.tsx:124]: if `props.period.gracePeriod` express the days, perhaps should it be converted in millisecond?
`props.period.endDate.getDate() + props.period.gracePeriod * 1000 * 60 * 60 * 24`

File: ts/features/bonus/bpd/store/reducers/details/activation/index.ts [hunk 3]
```diff
@@ -69,6 +69,7 @@ export const bpdEnabledSelector = (
 
 /**
  * Return the Iban that the user has entered to receive the cashback amount
+ * @return {RemoteValue<string | undefined, Error>}
  */
 export const bpdIbanSelector = createSelector(
```
Comment (Undermaken) [inline ts/features/bonus/bpd/store/reducers/details/activation/index.ts:74]: Expressing the type of generics accepted by createSelector you can inform typescript about the value type returned
Although it is more verbose, as a developer I prefer to know immediately what is the [type returned by the selector](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-function-return-type.md)

```typescript
export const bpdIbanSelector = createSelector<
  GlobalState,
  RemoteValue<string | undefined, Error>,
  RemoteValue<string | undefined, Error>
>
```

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/bonus/bpd/screens/details/components/summary/BpdSummaryComponent.tsx (hunk 1 above)
Comment (fabriziofff) [reply ts/features/bonus/bpd/screens/details/components/summary/BpdSummaryComponent.tsx:37]: Sure better!

File: ts/features/bonus/bpd/screens/details/components/summary/TextualSummary.tsx (hunk 2 above)
Comment (fabriziofff) [reply ts/features/bonus/bpd/screens/details/components/summary/TextualSummary.tsx:124]: This should be right according to the documentation https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setDate

File: ts/features/bonus/bpd/store/reducers/details/activation/index.ts (hunk 3 above)
Comment (fabriziofff) [reply ts/features/bonus/bpd/store/reducers/details/activation/index.ts]: Sure better!

File: ts/features/bonus/bpd/screens/details/components/summary/TextualSummary.tsx (hunk 2 above)
Comment (Undermaken) [reply ts/features/bonus/bpd/screens/details/components/summary/TextualSummary.tsx:124]: My fault! I didn't test it and I confused `getDate` with `getTime`

**Review by Undermaken — APPROVED** [review]

**Pull request merged**